C:/Users/ido.igip1/Anaconda3/DeepAdv3D/model_trainer_main.py:82: UserWarning: nn.init.xavier_uniform is now deprecated in favor of nn.init.xavier_uniform_.
  nn.init.xavier_uniform(m.weight.data)
C:/Users/ido.igip1/Anaconda3/DeepAdv3D/model_trainer_main.py:89: UserWarning: nn.init.xavier_uniform is now deprecated in favor of nn.init.xavier_uniform_.
  nn.init.xavier_uniform(m.weight.data)
CE:  tensor(27.0690, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #0: Batch 3/0] Train loss: 1.103243, Misclassified: [32/5]
CE:  tensor(38.2988, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #0: Batch 3/1] Train loss: 24.813385, Misclassified: [32/4]
CE:  tensor(28.5035, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #0: Batch 3/2] Train loss: 40.792249, Misclassified: [6/1]
CE:  tensor(29.8792, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #0: Batch 3/0] Validation loss: -5.886975, Misclassified: [15/2]
CE:  tensor(32.2682, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #1: Batch 3/0] Train loss: -22.214627, Misclassified: [32/0]
CE:  tensor(28.6981, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #1: Batch 3/1] Train loss: -20.843303, Misclassified: [32/4]
CE:  tensor(35.6428, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #1: Batch 3/2] Train loss: -20.692404, Misclassified: [6/0]
CE:  tensor(21.1889, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #1: Batch 3/0] Validation loss: -14.023394, Misclassified: [15/4]
CE:  tensor(31.9471, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #2: Batch 3/0] Train loss: -27.584520, Misclassified: [32/2]
CE:  tensor(27.0728, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #2: Batch 3/1] Train loss: -23.160370, Misclassified: [32/4]
CE:  tensor(26.1882, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #2: Batch 3/2] Train loss: -19.697822, Misclassified: [6/1]
CE:  tensor(30.7171, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #2: Batch 3/0] Validation loss: -27.121876, Misclassified: [15/2]
CE:  tensor(28.3014, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #3: Batch 3/0] Train loss: -25.993715, Misclassified: [32/4]
CE:  tensor(28.1625, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #3: Batch 3/1] Train loss: -25.939738, Misclassified: [32/7]
CE:  tensor(28.0733, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #3: Batch 3/2] Train loss: -23.103798, Misclassified: [6/0]
CE:  tensor(33.4045, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #3: Batch 3/0] Validation loss: -30.707897, Misclassified: [15/0]
CE:  tensor(30.5161, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #4: Batch 3/0] Train loss: -28.721754, Misclassified: [32/3]
CE:  tensor(32.6039, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #4: Batch 3/1] Train loss: -30.785497, Misclassified: [32/2]
CE:  tensor(35.0373, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #4: Batch 3/2] Train loss: -31.360441, Misclassified: [6/0]
CE:  tensor(37.4892, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #4: Batch 3/0] Validation loss: -35.040528, Misclassified: [15/0]
CE:  tensor(25.5549, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #5: Batch 3/0] Train loss: -23.962772, Misclassified: [32/3]
CE:  tensor(26.1590, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #5: Batch 3/1] Train loss: -24.588164, Misclassified: [32/8]
CE:  tensor(32.7668, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #5: Batch 3/2] Train loss: -29.366721, Misclassified: [6/0]
CE:  tensor(29.2054, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #5: Batch 3/0] Validation loss: -27.277777, Misclassified: [15/3]
CE:  tensor(26.3914, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #6: Batch 3/0] Train loss: -25.256004, Misclassified: [32/3]
CE:  tensor(30.3145, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #6: Batch 3/1] Train loss: -29.228401, Misclassified: [32/2]
CE:  tensor(19.9582, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #6: Batch 3/2] Train loss: -17.493266, Misclassified: [6/1]
CE:  tensor(25.5986, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #6: Batch 3/0] Validation loss: -24.252334, Misclassified: [15/1]
CE:  tensor(29.4353, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #7: Batch 3/0] Train loss: -28.604318, Misclassified: [32/3]
CE:  tensor(23.9820, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #7: Batch 3/1] Train loss: -23.124099, Misclassified: [32/7]
CE:  tensor(34.8674, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #7: Batch 3/2] Train loss: -32.703544, Misclassified: [6/1]
CE:  tensor(27.1326, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #7: Batch 3/0] Validation loss: -25.954639, Misclassified: [15/1]
CE:  tensor(30.5945, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #8: Batch 3/0] Train loss: -29.908832, Misclassified: [32/1]
CE:  tensor(30.3593, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #8: Batch 3/1] Train loss: -29.677730, Misclassified: [32/5]
CE:  tensor(25.1816, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #8: Batch 3/2] Train loss: -23.454664, Misclassified: [6/2]
CE:  tensor(30.4802, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #8: Batch 3/0] Validation loss: -29.516534, Misclassified: [15/2]
CE:  tensor(31.2412, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #9: Batch 3/0] Train loss: -30.653064, Misclassified: [32/1]
CE:  tensor(31.4162, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #9: Batch 3/1] Train loss: -30.810891, Misclassified: [32/2]
CE:  tensor(40.3580, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #9: Batch 3/2] Train loss: -38.857565, Misclassified: [6/0]
CE:  tensor(24.3320, device='cuda:0', grad_fn=<MulBackward0>)
[Epoch #9: Batch 3/0] Validation loss: -23.530671, Misclassified: [15/3]
Traceback (most recent call last):
  File "C:/Users/ido.igip1/Anaconda3/DeepAdv3D/model_trainer_main.py", line 125, in <module>
    train_ins.train()
  File "C:\Users\ido.igip1\Anaconda3\DeepAdv3D\src\deep_adv_3d\train_loop.py", line 103, in train
    self.evaluate(self.tensor_log_dir)
AttributeError: 'Trainer' object has no attribute 'evaluate'